test/ably/realtime/realtimechannel_vcdiff_test.py

```
@@ -0,0 +1,184 @@
+            await asyncio.wait_for(ably.connection.once_async(ConnectionState.CONNECTED), timeout=5)
+
+            channel = ably.channels.get('decode_failure_recovery', ChannelOptions(params={'delta': 'vcdiff'}))
+
+            # Monitor for attaching state changes
+            attaching_events = []
+
+            def on_attaching(state_change):
+                attaching_events.append(state_change)
+                # RTL18c - Check error code
+                if state_change.reason and state_change.reason.code:
+                    assert state_change.reason.code == 40018, "Check error code passed through per RTL18c"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)